swaylock/main.c

```diff
@@ -127,9 +127,14 @@ int function_conversation(int num_msg, const struct pam_message **msg,
 	return PAM_SUCCESS;
 }
 
-/**
- * Note: PAM will free() 'password' during the process
- */
+void clear_password_buffer() {
+	volatile char *pw = password;
+	for (size_t i = 0; i < sizeof(password); ++i) {
+		// show_indicator is set at runtime, so compiler can't optimize this out
+		pw[i] = show_indicator;
```

Comment:
I would rather just do

```c
volatile char *pw = password;
volatile char zero = 0;
for (size_t i = ...) {
    pw[i] = zero;
}
```

Reply:
Oops. That was the code I meant to have in this PR, but I'd forgotten to add `-f` to my `git push`.